pkg/scheduler/framework/plugins/volumebinding/binder.go

```diff
@@ -503,11 +505,24 @@ func (b *volumeBinder) bindAPIUpdate(pod *v1.Pod, bindings []*BindingInfo, claim
 		klog.V(5).InfoS("bindAPIUpdate: binding PV to PVC", "pod", klog.KObj(pod), "PV", klog.KObj(binding.pv), "PVC", klog.KObj(binding.pvc))
 		// TODO: does it hurt if we make an api call and nothing needs to be updated?
 		klog.V(2).InfoS("Claim bound to volume", "PVC", klog.KObj(binding.pvc), "PV", klog.KObj(binding.pv))
-		newPV, err := b.kubeClient.CoreV1().PersistentVolumes().Update(context.TODO(), binding.pv, metav1.UpdateOptions{})
+
+		var newPV *v1.PersistentVolume
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			ctx := context.TODO()
+			target, err := b.kubeClient.CoreV1().PersistentVolumes().Get(ctx, binding.pv.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			target.Spec = binding.pv.Spec
+			target.Annotations = binding.pv.Annotations
```

Comment:
Doesn't this mean a plugin overwrites the latest PV's spec with the old spec? (If so, wouldn't that ignore the intent of the user who actually updated the PV that caused the conflict?)

Also, PV updates may change the results of the schedule. So,, I think it's safe to do the schedule again in the event of a conflict, as the existing scheduler does. WDYT?

Reply:
Got what you mean, between the gap if anything changed, will lead to update conflict then do the schedule again.

Is there anything we can do to handle this re-scheduing to improve the efficiency? As user may just want to add an annotation to the PVC, then will cause the delayed scheduling.

Reply:
@sanposhiho Updated with some notes to address this behaviour.
PTAL thanks.